QgisModelBaker/gui/panel/mssql_config_panel.py

```diff
@@ -0,0 +1,175 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from qgis.PyQt.QtCore import pyqtSignal
+from qgis.PyQt.QtWidgets import (QLabel,
+                                 QGridLayout,
+                                 QLineEdit,
+                                 QCheckBox,
+                                 QComboBox)
+
+from .db_config_panel import DbConfigPanel
+from QgisModelBaker.utils.qt_utils import (
+    Validators,
+    NonEmptyStringValidator)
+from QgisModelBaker.utils.mssql_utils import get_odbc_drivers
+from QgisModelBaker.libili2db.globals import DbActionType
+
+
+class MssqlConfigPanel(DbConfigPanel):
+
+    notify_fields_modified = pyqtSignal(str)
+
+    def __init__(self, parent, db_action_type):
+        DbConfigPanel.__init__(self, parent, db_action_type)
+
+        lbl_odbc_driver = QLabel(self.tr("Odbc Driver"))
+        lbl_host = QLabel(self.tr("Host"))
+        lbl_instance = QLabel(self.tr("Instance"))
+        lbl_port = QLabel(self.tr("Port"))
+        lbl_database = QLabel(self.tr("Database"))
+        lbl_schema = QLabel(self.tr("Schema"))
+        lbl_user = QLabel(self.tr("User"))
+        lbl_password = QLabel(self.tr("Password"))
+
+        self.mssql_odbc_driver = QComboBox()
+
+        for item_odbc_driver in get_odbc_drivers():
+            self.mssql_odbc_driver.addItem(item_odbc_driver)
+
+        self.mssql_host_line_edit = QLineEdit()
+        self.mssql_host_line_edit.setPlaceholderText(self.tr("Database Hostname"))
+        self.mssql_host_line_edit.setText('localhost')
+
+        self.mssql_instance_line_edit = QLineEdit()
+        self.mssql_instance_line_edit.setPlaceholderText(self.tr("[Leave empty to use default instance]"))
+
+        self.mssql_port_line_edit = QLineEdit()
+        self.mssql_port_line_edit.setPlaceholderText(self.tr("[Leave empty to use dynamic or standard port 1433]"))
+
+        self.mssql_database_line_edit = QLineEdit()
+        self.mssql_database_line_edit.setPlaceholderText(self.tr("Database Name"))
+
+        self.mssql_schema_line_edit = QLineEdit()
+        self.mssql_schema_line_edit.setPlaceholderText(self.tr("[Leave empty to load all schemas in the database]"))
+
+        self.mssql_user_line_edit = QLineEdit()
+        self.mssql_user_line_edit.setPlaceholderText(self.tr("Database Username"))
+
+        self.mssql_password_line_edit = QLineEdit()
+        self.mssql_password_line_edit.setEchoMode(QLineEdit.Password)
+
+        layout = QGridLayout(self)
```

Comment:
but not in the case of layout
```suggestion
        layout = QGridLayout()
```

Reply:
That was moved to ui file df4dfa1d453b0e1bef7301bf17265a872a4cfc18